scripts/generator/graphql_generator/csharp/SDK/Android/AndroidNativeCheckout.AndroidPayEventReceiver.cs.erb

```diff
@@ -32,6 +35,49 @@ namespace <%= namespace %>.SDK.Android {
             UpdateShippingLine(shippingMethod, message);
         }
 
+        public void OnConfirmCheckout(string serializedMessage) {
+            var checkout = CartState.CurrentCheckout;
+            var message = NativeMessage.CreateFromJSON(serializedMessage);
+            var amount  = checkout.totalPrice();
+            var payment = new NativePayment(message.Content);
+            var tokenizedPaymentInput = new TokenizedPaymentInput(
+                amount: amount,
+                billingAddress: payment.BillingAddress,
+                idempotencyKey: payment.TransactionIdentifier,
+                paymentData: payment.PaymentData,
+                identifier: payment.Identifier,
+                type: "android_pay"
+            );
+
+            CartState.SetEmailAddress(payment.Email, (ShopifyError error) => {
+                if (error == null) {
+                    PerformCheckout(checkout, tokenizedPaymentInput, message);
+                } else {
+                    AndroidPayCheckoutResponse.Status status = AndroidPayCheckoutResponse.Status.Failure;
+                    message.Respond(new AndroidPayCheckoutResponse(status).ToJsonString());
+                }
+            });
+        }
+
+        private void PerformCheckout(Checkout checkout, TokenizedPaymentInput tokenizedPaymentInput,
+                NativeMessage message) {
+            CartState.CheckoutWithTokenizedPayment(tokenizedPaymentInput, (ShopifyError error) => {
+                AndroidPayCheckoutResponse.Status status = error == null
+                        ? AndroidPayCheckoutResponse.Status.Success
+                        : AndroidPayCheckoutResponse.Status.Failure;
+                message.Respond(new AndroidPayCheckoutResponse(status).ToJsonString());
+            });
+        }
+
+        private void SetFinalCheckoutFieldsForPayment(NativePayment payment, Checkout checkout,
```

Comment:
I don't see this being called anywhere. Shouldn't this be called just before performing the checkout?

Reply:
This is garbage and must be removed.

Reply:
Ah okay. Regarding your comment below, having everything be set at each point in the flow is the best approach. I just saw this method here and no references being made to it. All good 👍

Reply:
https://github.com/Shopify/unity-buy-sdk/pull/356/commits/4cc425f48b0758cf55b7724c2e4057b3b63f7db9